include/toolchain/gcc.h

```diff
@@ -380,4 +380,32 @@ do {                                                                    \
 	__asm__ __volatile__ ("" ::: "memory"); \
 } while (false)
 
+/** @brief Return larger value of two provided expressions.
+ *
+ * Macro ensures that expressions are evaluated only once.
+ *
+ * @note Macro has limited usage compared to the standard macro as it cannot be
+ *	 used:
+ *	 - to generate constant integer, e.g. __aligned(Z_MAX(4,5))
+ *	 - static variable, e.g. array like static u8_t array[Z_MAX(...)];
+ */
+#define Z_MAX(a, b) ({ \
+		/* random suffix to avoid naming conflict */ \
+		__typeof__ (a) _value_a__ = (a); \
```

Comment:
While C allows double-underscore to appear anywhere except the start of an identifier, in C++ a double underscore anywhere in the identifier invoked undefined behavior.  This header will be included in C++ compilations.  Please remove the second underscore at the end of these identifiers.

Reply:
done

Reply:
ping